xds/type/matcher/v3/string.proto

```diff
@@ -52,6 +52,25 @@ message StringMatcher {
     //
     // * *abc* matches the value *xyz.abc.def*
     string contains = 7 [(validate.rules).string = {min_len: 1}];
+
+    // The input string must have the pattern specified here.
+    // A pattern is defined as a set of values between "/"s.
+    // Each value can be a string literal or wildcard (*).
+    //
+    // Empty pattern match is not allowed.
+    //
+    // Note: There is overlap between regex and pattern.
+    // Pattern serves as a better user experience through simplified matching syntax.
+    //
+    // Examples:
+    //
+    // abc/*/* matches the value abc/eng/us
```

Comment:
Each example should be in bullet (line prefix with `*`). Also each of the actual pattern/value should be italic ([reference](https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#inline-markup-and-special-characters-e-g-bold-italic-verbatim)). See rendered output of previous docs: https://www.envoyproxy.io/docs/envoy/latest/api-v3/type/matcher/v3/string.proto

Unfortunately I don't think the cncf repo runs the docs rendering script in a PR. @phlax is this correct? How do we make sure these docs are rendered well?

Reply:
envoy pulls in the xds protos as a dep so when the dep is updated the newer protos are rendered

if you need to test a PR ahead of this you can create a PR in envoy and point the dep at the PR (commit hash for PR mostly will work as release version in `bazel/repository_locations.bzl`)